src/experimental/tools/videoThumbnailLoader/prepare_source_buffer.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  defer as observableDefer,
+  Observable,
+} from "rxjs";
+import { mergeMap } from "rxjs/operators";
+import openMediaSource from "../../../core/init/create_media_source";
+import { QueuedSourceBuffer } from "../../../core/source_buffers";
+
+/**
+ * Open the media source and create the queued source buffer.
+ * @param {HTMLVideoElement} elt
+ * @param {string} codec
+ * @returns {Observable}
+ */
+export default function prepareSourceBuffer(elt: HTMLVideoElement,
+                                            codec: string
+): Observable<QueuedSourceBuffer<Uint8Array>> {
+  return observableDefer(() => {
+    return openMediaSource(elt).pipe(
+      mergeMap((mediaSource) => {
+        return new Observable<QueuedSourceBuffer<Uint8Array>>((obs) => {
+          const sourceBuffer = mediaSource.addSourceBuffer(codec);
+          const queuedSourceBuffer =
+            new QueuedSourceBuffer<Uint8Array>("video", codec, sourceBuffer);
+          obs.next(queuedSourceBuffer);
+          return;
+        });
```

Comment:
There are two things here:
 1. Why a mergeMap considering that there are no asynchronicity and no EMPTY, it could just as well be a `map` (or you could just return an observable.of)?
  2. You forgot to complete when calling next, which could lead to leaks and/or never-ending observables.